repository-meta-analyzer/src/main/java/org/dependencytrack/processor/MetaAnalyzerProcessor.java

```diff
@@ -67,95 +67,37 @@ public void process(final FixedKeyRecord<PackageURL, AnalysisCommand> record) {
-                    var integrityMeta = fetchIntegrityMeta(analyzer, repository, record.value());
-                    if (integrityMeta != null) {
-                        var metaBuilder = org.dependencytrack.proto.repometaanalysis.v1.IntegrityMeta.newBuilder();
-                        Optional.ofNullable(integrityMeta.getMd5()).ifPresent(hash -> metaBuilder.setMd5(hash));
-                        Optional.ofNullable(integrityMeta.getSha1()).ifPresent(hash -> metaBuilder.setSha1(hash));
-                        Optional.ofNullable(integrityMeta.getSha256()).ifPresent(hash -> metaBuilder.setSha256(hash));
-                        Optional.ofNullable(integrityMeta.getSha512()).ifPresent(hash -> metaBuilder.setSha512(hash));
-                        Optional.ofNullable(integrityMeta.getMetaSourceUrl()).ifPresent(url -> metaBuilder.setMetaSourceUrl(url));
-                        Optional.ofNullable(integrityMeta.getCurrentVersionLastModified()).ifPresent(date ->
-                                metaBuilder.setCurrentVersionLastModified(Timestamp.newBuilder()
-                                        .setSeconds(date.getTime() / 1000)));
-                        resultBuilder.setIntegrityMeta(metaBuilder);
-                        cacheResult(cacheKeyWithVersion, resultBuilder.build());
-                    }
-                }
-                if (record.value().getFetchMeta().equals(FetchMeta.FETCH_META_INTEGRITY_DATA)) {
-                    // forward result for only integrity meta
-                    context().forward(record
-                            .withValue(resultBuilder.build())
-                            .withTimestamp(context().currentSystemTimeMs()));
-                    return;
-                }
+        }
+
+        if (analysisCommand.getFetchMeta().equals(FetchMeta.FETCH_META_INTEGRITY_DATA)
+                || analysisCommand.getFetchMeta().equals(FetchMeta.FETCH_META_INTEGRITY_DATA_AND_LATEST_VERSION)) {
+            resultBuilder = performIntegrityMeta(applicableRepositories, analyzer, analysisCommand, purl, resultBuilder);
+            if (analysisCommand.getFetchMeta().equals(FetchMeta.FETCH_META_INTEGRITY_DATA)) {
+                // forward result for only integrity meta
+                context().forward(record
+                        .withValue(resultBuilder.build())
+                        .withTimestamp(context().currentSystemTimeMs()));
+                return;
             }
-            // forward result for both latest version and integrity meta
-            context().forward(record
-                    .withValue(resultBuilder.build())
-                    .withTimestamp(context().currentSystemTimeMs()));
-            return;
         }
-        // Produce "empty" result in case no repository did yield a satisfactory result.
+        // forward result with component even when no satisfactory results were yielded.
```

Comment:
here actually the result trickling down also seems to be the one that has both integrity meta and repo meta information: FETCH_META_INTEGRITY_DATA_AND_LATEST_VERSION this case. So maybe update this comment